src/prezzemolo/graph.py

```diff
@@ -87,6 +110,42 @@ def breadth_first_search(self, start: Vertex[ValueType], end: Vertex[ValueType])
         # start and end are type checked inside _breadth_first_search()
         return self._breadth_first_search(start, end, vertex_2_parent=None)
 
+    def _dijkstra_search(
+        self, start: Vertex[ValueType], end: Optional[Vertex[ValueType]], vertex_2_parent: Optional[Dict[Vertex[ValueType], Optional[Vertex[ValueType]]]] = None
+    ) -> bool:
+        distance: Dict[Vertex[ValueType], float] = {v: float("inf") for v in self.__vertexes}
+        distance[start] = 0.0
+        remaining_vertexes: PriorityQueue[_ShortestDistance[ValueType]] = PriorityQueue()
+        remaining_vertexes.put(_ShortestDistance(distance=0.0, vertex=start))
+        visited: Set[Vertex[ValueType]] = set()
+
+        while not remaining_vertexes.empty():
+            next_vertex: _ShortestDistance[ValueType] = remaining_vertexes.get()
+            current_distance: float = next_vertex.distance
+            current_vertex: Vertex[ValueType] = next_vertex.vertex
```

Comment:
In these 3 lines `next_vertex` and `current_vertex` are a little confusing because one is current, the other in next but they refer to different parts of the same object. I suggest renaming `next_vertex` -> `shortest_distance`.